pkg/controller/baremetalhost/action_result.go

```diff
@@ -14,6 +14,25 @@ type actionResult interface {
 	Dirty() bool
 }
 
+// actionContinueNoWrite is a result indicating that the current action is still
+// in progress, and that the resource should remain in the same provisioning
+// state without writing the status
+type actionContinueNoWrite struct {
+	delay time.Duration
```

Comment:
Instead of duplicating the content of actionContinue (and the `Result()` method), we can just compose it:
```
type actionContinueNoWrite struct {
    actionContinue
}
```
Then you only need to override the `Dirty()` method.

Reply:
nice, done.